homekit/StateManager.js

```diff
@@ -1,623 +1,1034 @@
+		smartModeState.lowTemperatureWebhook = null
+
+		if (device.state.mode === 'COOL') {
+			smartModeState.highTemperatureThreshold = device.state.targetTemperature + (device.usesFahrenheit ? 1.8 : 1)*climateReactAutoSetupMultiplier + climateReactAutoSetupOffset
+			smartModeState.highTemperatureState = {
+				on: true,
+				targetTemperature: device.state.targetTemperature,
+				temperatureUnit: device.temperatureUnit,
+				mode: device.state.mode,
+				fanSpeed: device.state.fanSpeed,
+				swing: device.state.verticalSwing,
+				horizontalSwing: device.state.horizontalSwing,
+				light: device.state.light ? 'on' : 'off'
+			}
 
-	// NOTE: device.state is of "type" StateHandler. When one of its properties is "set" (e.g. device.state.<property> = <val>),
-	//       that's where we actually send commands to the appropriate Sensibo devices. If a property is not set, the aformentioned
-	//       code will not execute and the changes would not take effect.
-	//
-	//       For example, if we set a property of smartMode directly, e.g. device.state.smartMode.enabled = true, StateHandler's
-	//       setter will not get called and so any changes will not take effect. This is why we MUST update a device's property as
-	//       a whole, and do it only once (otherwise's the setter will get called multiple times which will send repeated commands
-	//       to the Sensibo devices).
-	device.state.smartMode = smartModeState
-}
+			smartModeState.lowTemperatureThreshold = device.state.targetTemperature - (device.usesFahrenheit ? 1.8 : 1)*climateReactAutoSetupMultiplier + climateReactAutoSetupOffset
+			smartModeState.lowTemperatureState = {
+				on: false,
+				targetTemperature: device.state.targetTemperature,
+				temperatureUnit: device.temperatureUnit,
+				mode: device.state.mode,
+				fanSpeed: device.state.fanSpeed,
+				swing: device.state.verticalSwing,
+				horizontalSwing: device.state.horizontalSwing,
+				light: device.state.light ? 'on' : 'off'
+			}
+		} else if (device.state.mode === 'HEAT') {
+			smartModeState.highTemperatureThreshold = device.state.targetTemperature + (device.usesFahrenheit ? 1.8 : 1)*climateReactAutoSetupMultiplier + climateReactAutoSetupOffset
+			smartModeState.highTemperatureState = {
+				on: false,
+				targetTemperature: device.state.targetTemperature,
+				temperatureUnit: device.temperatureUnit,
+				mode: device.state.mode,
+				fanSpeed: device.state.fanSpeed,
+				swing: device.state.verticalSwing,
+				horizontalSwing: device.state.horizontalSwing,
+				light: device.state.light ? 'on' : 'off'
+			}
 
-// TODO: perhaps make this a class?
-module.exports = (device, platform) => {
-	Characteristic = platform.api.hap.Characteristic
-	const log = platform.log
-	const enableClimateReactAutoSetup = platform.enableClimateReactAutoSetup
+			smartModeState.lowTemperatureThreshold = device.state.targetTemperature - (device.usesFahrenheit ? 1.8 : 1)*climateReactAutoSetupMultiplier + climateReactAutoSetupOffset
```

Comment:
Should this be:
```suggestion
			smartModeState.lowTemperatureThreshold = device.state.targetTemperature - ((device.usesFahrenheit ? 1.8 : 1) * climateReactAutoSetupMultiplier + climateReactAutoSetupOffset)
```
Otherwise wouldn't the Offset _increase_ rather than decrease the threshold?

Reply:
This is by design, the purpose of this is to allow for an offset in the **temperature range**, e.g. when setting to cool 25C, instead of keeping the temperature between 24 and 26, setting an offset of 1, you will keep the temperature between 25 and 27. I actually use this exactly for that purpose.